### PR TITLE
Create two classes and redifine Clone method

### PR DESCRIPTION
**Pull request by fyarkiy**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        clone().brand = brand;
+        clone().model = model;
+        clone().engine = engine;
+        clone().numberOfDoors = numberOfDoors;
+        clone().version = version;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java:23]: використай конструктор в даному випадку:
new Car(brand, model...)
переглянь чи точно скопіюється engine з поточною реалізацією

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+        this.power = power;
+        this.modelName = modelName;
+        this.electric = electric;
+        this.yearModel = yearModel;
+        this.type = type;
+    }
+
+    @Override
+    public Engine clone() throws CloneNotSupportedException {
+        clone().power = power;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Engine.java:20]: the same

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (fyarkiy) [reply src/main/java/core/basesyntax/Car.java]: Didn't work - I've made changes. Thanks!

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        brand = brand;
+        model = model;
+        version = version;
+        numberOfDoors = numberOfDoors;
+        engine = engine;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Car.java:24]: this makes no sense

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+        return new Car(brand, model, version, numberOfDoors, engine);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Car.java:25]: you do not clone engine here
try creating a car, then clone it with `car.clone()`
and then try changing engine type on the original car

the engine type of the cloned car should not change

File: src/main/java/core/basesyntax/Engine.java [hunk 5]
```diff
@@ -1,5 +1,32 @@
...
+        this.modelName = modelName;
+        this.electric = electric;
+        this.yearModel = yearModel;
+        this.type = type;
+    }
+
+    public Engine(int power, String modelName) {
+        this.power = power;
+        this.modelName = modelName;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Engine.java:21]: do we use this constructor in our code?

File: src/main/java/core/basesyntax/Engine.java [hunk 6]
```diff
@@ -1,5 +1,32 @@
...
+
+    @Override
+    public Engine clone() throws CloneNotSupportedException {
+        power = power;
+        modelName = modelName;
+        electric = electric;
+        yearModel = yearModel;
+        type = type;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Engine.java:29]: this is redundant as well
you can remove these lines

File: src/main/java/core/basesyntax/Engine.java [hunk 7]
```diff
@@ -1,5 +1,32 @@
...
+        return (Engine) new Engine(power, modelName, electric, yearModel, type);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Engine.java:30]: do we need to cast here?

File: src/main/java/core/basesyntax/Car.java (hunk 3 above)
Comment (fyarkiy) [reply src/main/java/core/basesyntax/Car.java]: Originally the task was  - not to use super.clone(). I have changed it now

File: src/main/java/core/basesyntax/Car.java [hunk 8]
```diff
@@ -1,5 +1,38 @@
...
+        this.brand = brand;
+        this.model = model;
+        this.version = version;
+        this.numberOfDoors = numberOfDoors;
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        return (Car) super.clone();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Car.java:20]: please return to your previous implementation here
you need to create a new Car here and return it
you can use constructor, but you should pass not the same Engine object - but its clone

Comment (fyarkiy) [reply src/main/java/core/basesyntax/Car.java:25]: I have made changes but still don't get what is the point to do like this? Do we want CarClone (during creation) to have different Engines (references to different objects) in both cars? Then I didn't achieve it and don't understand how to do it:(.

**Pull request closed without merging**